src/schema/schema.rs

```diff
@@ -169,12 +173,30 @@ pub struct Schema<Root: SchemaRoot> {
     pub transitions: TinyOrdMap<TransitionType, TransitionSchema>,
 
     /// Type system
-    // TODO: Commit to type system id, not the type system itself
     pub type_system: TypeSystem,
     /// Validation code.
     pub script: Script,
 }
 
+impl<Root: SchemaRoot> CommitEncode for Schema<Root> {
+    type CommitmentId = SchemaId;
+
+    fn commit_encode(&self, e: &mut CommitEngine) {
+        e.commit_to(&self.ffv);
+        e.commit_to(&self.subset_of.as_ref().map(Root::schema_id));
+
+        e.commit_to(&self.global_types);
+        e.commit_to(&self.owned_types);
+        e.commit_to(&self.valency_types);
+        e.commit_to(&self.genesis);
+        e.commit_to(&self.extensions);
+        e.commit_to(&self.transitions);
+
+        e.commit_to(&self.type_system.id());
+        e.commit_to(&self.script);
```

Comment:
All commit_to parameters need be named type, to avoid the error: `commit encoder can commit only to named types` 

```suggestion
        e.commit_to(&self);
```

Reply:
Nope, we will be back to the round 1 with creating commitments to rangeproofs and not having deterministic transaction ids. 

We need here to make named types for all the collections... or change the way CommitEngine works in `commit_verify` crate. I will try to do this ASAP. 

... so tired of these commitments ...

Reply:
> Nope, we will be back to the round 1 with creating commitments to rangeproofs and not having deterministic transaction ids.
> 
> We need here to make named types for all the collections... or change the way CommitEngine works in `commit_verify` crate. I will try to do this ASAP.
> 
> ... so tired of these commitments ...

Yes, I comment in another [review](https://github.com/RGB-WG/rgb-std/pull/148#discussion_r1492989668). I believe it makes more sense to use a wrapper, to avoid this in the future.